apps/console/src/features/roles/components/role-list-updated.tsx

```diff
@@ -0,0 +1,341 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AccessControlConstants, Show } from "@wso2is/access-control";
+import { RoleConstants } from "@wso2is/core/constants";
+import { hasRequiredScopes } from "@wso2is/core/helpers";
+import {
+    IdentifiableComponentInterface,
+    LoadableComponentInterface,
+    RoleListInterface,
+    RolesInterface
+} from "@wso2is/core/models";
+import {
+    AnimatedAvatar,
+    AppAvatar,
+    ConfirmationModal,
+    DataTable,
+    EmptyPlaceholder,
+    LinkButton,
+    PrimaryButton,
+    TableActionsInterface,
+    TableColumnInterface
+} from "@wso2is/react-components";
+import React, { ReactElement, ReactNode, SyntheticEvent, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import { Header, Icon, Label, SemanticICONS } from "semantic-ui-react";
+import {
+    AppConstants,
+    AppState,
+    FeatureConfigInterface,
+    getEmptyPlaceholderIllustrations,
+    history
+} from "../../core";
```

Comment:
Shall we import from individual files rather than the index file? Since we have a plan to get rid of exporting through index files. Let's try to follow this in new components.
Ex:
```
import { AppConstants } from "../../core/constants/app-constants";
import { FeatureConfigInterface } from "../../core/models/config";
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4280/commits/31b88ad29860876b65b8843de775b5b7a096cd5e